core/src/main/java/com/linecorp/armeria/server/ServerBuilder.java

```diff
@@ -1134,6 +1134,17 @@ ServerBuilder decorator(Function<T, R> decorator) {
         return this;
     }
 
+    /**
+     * Decorates all {@link Service}s with the specified {@code decorator}.
+     *
+     * @param decoratingServiceFunction the {@link DecoratingServiceFunction} that decorates a {@link Service}.
+     */
+    public ServerBuilder decorator(
+            DecoratingServiceFunction<HttpRequest, HttpResponse> decoratingServiceFunction) {
+        requireNonNull(decoratingServiceFunction, "decorator");
+        return decorator(service -> (ctx, req) -> decoratingServiceFunction.serve(service, ctx, req));
```

Comment:
You must use `FunctionalDecoratingService` because otherwise the methods like `as()` and `serviceAdded()` will not be delegated properly.